hawc/apps/materialized/models.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.core.cache import cache
+from django.db import connection, models, transaction
+
+from . import managers
+
+
+class MaterializedViewModel(models.Model):
+    """
+    Base class for materialized views.
+    Django does not manage view creation automatically; that is handled by custom SQL queries in migrations.
+    And since these are views, any foreign keys on these models are not "true" foreign keys, so on_delete
+    should be kept as DO_NOTHING to prevent Django from enforcing foreign key constraints.
+    """
+
+    class Meta:
+        abstract = True
+        managed = False
+
+    @classmethod
+    def set_refresh_view_cache(cls, refresh: bool):
```

Comment:
again, while correct, view by itself is confusing a bit. consider renaming to

- set_refresh_flag
- get_refresh_flag
- refresh_if_flagged